app/code/Magento/Catalog/Controller/Adminhtml/Product/Validate.php

```diff
@@ -73,14 +95,20 @@ public function __construct(
         \Magento\Catalog\Model\Product\Validator $productValidator,
         \Magento\Framework\Controller\Result\JsonFactory $resultJsonFactory,
         \Magento\Framework\View\LayoutFactory $layoutFactory,
-        \Magento\Catalog\Model\ProductFactory $productFactory
+        \Magento\Catalog\Model\ProductFactory $productFactory,
+        ProductUrlPathGenerator $productUrlPathGenerator,
+        UrlFinderInterface $urlFinder,
+        GetStoresListByWebsiteIds $getStoresListByWebsiteIds
```

Comment:
added constructor parameters should be nullable and should be initialized inside constructor

Reply:
Thank you, fixed.